src/Symfony/Component/Translation/MessageCatalogueInterface.php

```diff
@@ -34,7 +34,7 @@ public function getLocale();
      *
      * @return array An array of domains
      */
-    public function getDomains();
+    public function getDomains(/* bool $excludeIntlDomains = true */);
```

Comment:
that API is too specific to a message format
how do we tell translators which format they should use?
Didn't we push this info somewhere for the provider to know?
/cc @welcoMattic wdyt?

Reply:
Yes, it's the tricky part of this fix: how to tell know, from the Provider point of view, if we have to work with intl domain or not? Especially when there is not existing translation file in the application. 

The only feasible solution I see is to ask explicitly to users if they need intl domain or not at Command level (with an option). Something like:

```php
# This will write in messages+intl-icu.fr.xlf
bin/console translation:pull loco --domains messages --intl-icu

# This will write in messages.fr.xlf
bin/console translation:pull loco --domains messages --not-intl-icu
```

And `--intl-icu` will be the default.

I don't see any other way to "guess" the intl format.

Reply:
What if we use both ICU and not ICU translations?

Ideally `bin/console translation:pull loco --domains messages` should pull translations from both `messages` (if available) and `messages+intl-icu` (if available), but that's an other issue I think.

Reply:
I consider that the domain is always `messages`, and the `+intl-icu` suffix is internal to Symfony, it should never get pushed to the Providers.

Reply:
Well, if we don't push the `+intl-icu` suffix to the provider, we still need to push a way to know which pluralization format is used by the translations. Otherwise, we are breaking the feature.

This `+intl-icu` suffix is internal to the Symfony TranslatorInterface, because consumers of the interface don't care about that format. However, it is not internal to the loading side of the component (which would defeat the purpose entirely) and providers are integrating with that side of the component as they are about managing translation resources.

Reply:
I agree we could push the suffix to the provider so that both the command and the human translator can know

Reply:
Ok, I agree too. I suppose it is not a BC as the feature is actually broken, right @nicolas-grekas?

Reply:
yep

Reply:
To resume, now we push the complete domain name (with suffix in such case), so translators (people) will be aware of the format they should use. Ok for you?